sparkle/tools/solver_wrapper_parsing.py

```diff
@@ -0,0 +1,73 @@
+"""This module provides tools for the argument parsing for solver wrappers."""
+from pathlib import Path
+import ast
+from typing import Any
+
+
+def parse_commandline_dict(args: list[str]) -> dict:
+    """Parses a commandline dictionary to the object."""
+    dict_str = " ".join(args)
+    # Remove white space
+    dict_str = dict_str.replace(" ", "")
+    # Remove all quotes to avoid double strings
+    dict_str = dict_str.replace('"', "").replace("'", "")
+    # Place the quotes only there where needed
+    dict_str = dict_str.replace("{", "{'").replace(":", "':'")
+    dict_str = dict_str.replace("[", "['").replace("]", "']")
+    dict_str = dict_str.replace(",", "','").replace("}", "'}")
+    # Undo quotes at the start and end of lists
+    dict_str = dict_str.replace("':'[", "':[").replace("]','", "],'")
+
+    return ast.literal_eval(dict_str)
+
+
+def parse_solver_wrapper_args(args: list[str]) -> dict[Any]:
+    """Parse the arguments passed to the solver wrapper.
+
+    Args:
+        args: a list of arguments passed via the command line. It is ensured by Sparkle
+            that this list contains certain keys such as `solver_dir`.
+
+    Returns:
+        A dictionary mapping argument names to their currently held values.
+    """
+    args_dict = parse_commandline_dict(args)
+
+    # Some data needs specific formatting
+    args_dict["solver_dir"] = Path(args_dict["solver_dir"])
+    args_dict["instance"] = Path(args_dict["instance"])
+    args_dict["seed"] = int(args_dict["seed"])
+
+    return args_dict
+
+
+def get_solver_call_params(args_dict: dict) -> list[str]:
+    """Gather the additional parameters for the solver call.
+
+    Args:
+        args_dict: dictionary mapping argument names to their currently held values
+
+    Returns:
+        A list of parameters for the solver call
+    """
+    params = []
+    if "config_path" in args_dict:
```

Comment:
It would be nice if this is already handled when parsing the original arguments dictionary, e.g. when it is first parsed, check if that config_path key is in there and if so try to read from it and place the variables in the dictionary. Then in this step all you need to do is the conversion from dictionary to a CLI arguments string

Reply:
So get_solver_call_params should not be a separate method?